src/Concordium/Client/Runner.hs

```diff
@@ -3664,174 +3620,74 @@ processLegacyCmd action backend =
         printSuccess
     GetIdentityProviders block ->
       withClient backend $
-        readOrFailM Best block >>=
+        readBlockHashOrDefault Best block >>=
         getIdentityProvidersV2 >>=
-        printResponseValueAsJSON'
+        printResponseValueAsJSON
     GetAnonymityRevokers block ->
       withClient backend $
-        readOrFailM Best block >>=
+        readBlockHashOrDefault Best block >>=
         getAnonymityRevokersV2 >>=
-        printResponseValueAsJSON'
+        printResponseValueAsJSON
     GetCryptographicParameters block ->
       withClient backend $
-        readOrFailM Best block >>=
+        readBlockHashOrDefault Best block >>=
         getCryptographicParametersV2 >>=
-        printResponseValueAsJSON'
+        printResponseValueAsJSON
   where
     -- |Print the response value under the provided mapping,
-    -- or fail if the response contained an error.
+    -- or fail with an error message if the response contained
+    -- an error.
+    printResponseValue :: (MonadIO m, Show b)
+      => (a -> b)
+      -> GRPCResult (Either String a)
+      -> m ()
     printResponseValue f res =
       getResponseValueOrFail' f res >>= liftIO . print
 
-    -- |Print the response value under the provided mapping
-    -- as JSON, or fail if the response contained an error.
-    printResponseValueAsJSON res f = do
-      v <- getResponseValueOrFail' f res
+    -- |Print the response value as JSON, or fail with an error
+    -- message if the response contained an error.
+    printResponseValueAsJSON :: (MonadIO m, ToJSON a)
+      => GRPCResult (Either String a)
+      -> m ()
+    printResponseValueAsJSON res = do
+      v <- getResponseValueOrFail res
       printJSONValues . toJSON $ v
 
-    -- |Print the response value as JSON, or fail if the response
-    -- contained an error.
-    printResponseValueAsJSON' res =
-      printResponseValueAsJSON res id
-
-    printSuccess (Left x)  = liftIO $ putStrLn $ "FAIL: " <> x
-    printSuccess (Right _) = liftIO $ putStrLn "OK"
+    -- |Print result of a query with side-effects.
+    printSuccess (Left x)  = liftIO $ logError [[i|FAIL: #{x}|]]
```

Comment:
`logError` prefixes `Error:` on its own, so `FAIL:` seems redundant.

Reply:
I agree.

Reply:
If we want to retain the interface as much as possible should I perhaps just use `printStrLn` as before here then?

Reply:
It is not about retaining the interface here. Here the message is just ugly with Error: FAIL. Error indicates failure already.

Reply:
Right, but doesn't the change from:

`putStrLn $ "FAIL: " <> x`
 
 to

`logError [[i|whatever: #{x}|]]`
 
change the output and hence the interface? I was just wondering if we should do as before and do `putStrLn` or use `logFatal` in this case?

Reply:
I'll use `putStrLn` here for now to preserve the output. Let me know if it should be changed.